plugins/cluster-status-backend/src/service/router.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { errorHandler } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+import { StatusCheck } from '../statusCheck/StatusCheck';
+import { ClusterDetails } from '../types/types';
+
+export interface RouterOptions {
+  logger: Logger;
+  config: Config;
+}
+
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const { logger } = options;
+  const { config } = options;
+
+  const unavaliableCluster: ClusterDetails = {
+    status: {
+      avaliable: false,
+      reason: 'ACM cluster unreachable'
+    }
+  }
+
+  const statusCheck = new StatusCheck(config, logger);
+
+  const router = Router();
+  router.use(express.json());
+
+  router.get('/status/:clusterName', (request, response) => {
+    const clusterName = request.params.clusterName
```

Comment:
This can be unpacked directly:

```suggestion
  router.get('/status/:clusterName', ({ params: { clusterName }}, response) => {
```

Reply:
Resolved in the latest commit.